src/main/java/com/eerussianguy/blazemap/feature/maps/MapRenderer.java

```diff
@@ -432,17 +436,40 @@ private void generateMapTileAsync(NativeImage texture, TileResolution resolution
     private void generateMapTile(NativeImage texture, TileResolution resolution, int textureW, int textureH, int cornerXOffset, int cornerZOffset, int regionIndexX, int regionIndexZ) {
         for(BlazeRegistry.Key<Layer> layer : visible) {
             if(layer.value() instanceof FakeLayer) return;
+
```

Comment:
Would appreciate if somebody can check my maths on the refactor below. Looks fine to me in-game, but a sanity check to make sure I haven't missed an edge case would be super helpful!

All the same calculations as before should be happening, but anything that shouldn't change loop-to-loop has been extracted to reduce:
a) the amount of time in each loop, and
b) the number of loops total through just not running those that noop.

Reply:
Am not sure _how much_ time is objectively saved per tick by these changes, since it's very hard to get a baseline without so much noise from other factors that it overwhelms the measurement, but it makes my optimisation-loving brain go brrr 😛